### PR TITLE
unarchiveObjectWithData: and archivedDataWithRootObject: are deprecated

### PR DESCRIPTION
**Pull request by woshimaliang**
migrate code to new apis.

### CODE REVIEW

File: Carthage/Checkouts/PINCache/Source/PINDiskCache.h [hunk 1]
```diff
@@ -13,16 +13,20 @@ NS_ASSUME_NONNULL_BEGIN
 @class PINDiskCache;
 @class PINOperationQueue;
 
-extern NSString * const PINDiskCacheErrorDomain;
-extern NSErrorUserInfoKey const PINDiskCacheErrorReadFailureCodeKey;
-extern NSErrorUserInfoKey const PINDiskCacheErrorWriteFailureCodeKey;
+extern NSString * const PINDiskCacheLogErrorDomain;
+extern NSErrorUserInfoKey const PINDiskCacheLogErrorReadFailureCodeKey;
+extern NSErrorUserInfoKey const PINDiskCacheLogErrorWriteFailureCodeKey;
```
Comment (jparise) [inline Carthage/Checkouts/PINCache/Source/PINDiskCache.h:18]: Won't changing these names will break backwards compatibility?

File: Carthage/Checkouts/PINCache/Source/PINDiskCache.m [hunk 2]
```diff
@@ -343,7 +339,16 @@ - (PINDiskCacheSerializerBlock)defaultSerializer
...
 {
     return ^id(NSData * data, NSString *key){
-        return [NSKeyedUnarchiver unarchiveObjectWithData:data];
+        if (@available(iOS 11.0, macOS 10.13, tvOS 11.0, watchOS 4.0, *)) {
+            NSError *error = nil;
+            NSKeyedUnarchiver *unarchiver = [[NSKeyedUnarchiver alloc] initForReadingFromData:data error:&error];
+            PINDiskCacheLogError(error);
+            unarchiver.requiresSecureCoding = NO;
+            id obj = [unarchiver decodeObjectForKey:NSKeyedArchiveRootObjectKey];
+            return obj;
```
Comment (jparise) [inline Carthage/Checkouts/PINCache/Source/PINDiskCache.m:348]: Just `return` without the intermediate `obj` variable?

(other occurrences of this below)

File: Carthage/Checkouts/PINCache/Source/PINDiskCache.h (hunk 1 above)
Comment (woshimaliang) [reply Carthage/Checkouts/PINCache/Source/PINDiskCache.h:18]: (these are unintended changes but by find/replace...)

good catch.  I don't need other changes either.

File: Carthage/Checkouts/PINCache/Source/PINDiskCache.h [hunk 3]
```diff
@@ -19,8 +19,8 @@ extern NSErrorUserInfoKey const PINDiskCacheErrorWriteFailureCodeKey;
 extern NSString * const PINDiskCachePrefix;
 
 typedef NS_ENUM(NSInteger, PINDiskCacheError) {
-  PINDiskCacheErrorReadFailure = -1000,
-  PINDiskCacheErrorWriteFailure = -1001,
+    PINDiskCacheErrorReadFailure = -1000,
+    PINDiskCacheErrorWriteFailure = -1001,
```
Comment (jparise) [inline Carthage/Checkouts/PINCache/Source/PINDiskCache.h:23]: I think you could revert this formatting change, too.

**Review by jparise — APPROVED** [review]

File: Carthage/Checkouts/PINCache/Source/PINDiskCache.h (hunk 3 above)
Comment (woshimaliang) [reply Carthage/Checkouts/PINCache/Source/PINDiskCache.h:23]: i could but previously they're not right indented?

Comment (jparise) [reply Carthage/Checkouts/PINCache/Source/PINDiskCache.h:23]: It's fine then. It's just not related to the rest of this change.

**Review by bolsinga — APPROVED** [review]

**Pull request merged**